components/bt/host/bluedroid/api/include/api/esp_bt_defs.h

```diff
@@ -166,15 +166,24 @@ typedef enum {
 /// Bluetooth peer irk
 #define ESP_PEER_IRK_LEN    16
 
-/// Bluetooth device address
-typedef uint8_t esp_bd_addr_t[ESP_BD_ADDR_LEN];
+// Bluetooth device address
+typedef uint8_t esp_bd_addr_t[ESP_BD_ADDR_LEN]; /*!< Bluetooth device address. Typically 6 bytes (48 bits). */
 
-/// BLE device address type
+/**
+ * @brief LE device address type
+ *        - More details in Spec BLUETOOTH CORE SPECIFICATION Version 5.4 Vol 4, Part E.
```

Comment:
1 Public Device Address is a 48-bit extended unique identifier and will not change after factory set, It can be read using esp_bt_dev_get_address().
2 Static random address wil not change once initialized using `esp_ble_gap_set_rand_addr()` until the device is power cycled.
3 Non-resolvable private address (NRPA) can be changed anytime using `esp_ble_gap_set_rand_addr()`.
4 Resolvable private address (RPA)
- The Controller use Local IRK to generate a RPA as own address.
- The Controller use the peer address to locate the corresponding Local IRK in the resolving list.
- If you want to using unique Local IRK for different peer device, enable the config `CONFIG_BT_BLE_SMP_ID_RESET_ENABLE` and bond with peer device, then specify the peer address.
- If you want to used the fixed Local IRK, set the peer address to all zero.